spec/controllers/users/nbp_wallet_controller_spec.rb

```diff
@@ -0,0 +1,211 @@
+# frozen_string_literal: true
+
+require 'rails_helper'
+
+RSpec.describe Users::NbpWalletController do
+  render_views
+
+  let(:mockfiles_path) { 'spec/controllers/users/enmeshed_connector_mocks' }
+  let(:connector_api_url) { "#{Settings.omniauth.nbp.enmeshed.connector_url}/api/v2" }
+
+  before do
+    stub_request(:get, "#{connector_api_url}/Account/IdentityInfo")
+      .to_return(body: File.read("#{mockfiles_path}/get_enmeshed_address.json"))
+
+    stub_request(:post, "#{connector_api_url}/Attributes")
+      .to_return(body: File.read("#{mockfiles_path}/display_name_created.json"))
+
+    stub_request(:post, "#{connector_api_url}/RelationshipTemplates/Own")
+      .to_return(body: File.read("#{mockfiles_path}/relationship_template_created.json"))
+
+    session[:omniauth_provider_uid] = 'example_uid'
+  end
+
+  describe 'GET #connect' do
+    subject(:get_request) { get :connect }
+
+    context 'when connector is down' do
+      before { stub_request(:get, "#{connector_api_url}/Account/IdentityInfo").to_timeout }
+
+      it 'redirects to registration page' do
+        get_request
+        expect(response).to redirect_to(new_user_registration_path)
+      end
+    end
+
+    context 'without errors' do
+      it 'sets correct template url' do
+        get_request
+        expect(assigns(:template_url)).to eq('nmshd://tr#relationship_template_example_truncated_reference')
+      end
+    end
+  end
+
+  describe 'GET #relationship_status' do
+    subject(:get_request) { get :relationship_status }
+
+    before { session[:relationship_template_id] = 'example_relationship_template_id' }
+
+    context 'when connector is down' do
+      before { stub_request(:get, "#{connector_api_url}/Relationships?status=Pending") }
+
+      it 'redirects to connect page' do
+        get_request
+        expect(response).to redirect_to(nbp_wallet_connect_users_path)
+      end
+    end
+
+    context 'when no relationship created yet' do
+      before do
+        stub_request(:get, "#{connector_api_url}/Relationships?status=Pending")
+          .to_return(body: File.read("#{mockfiles_path}/no_relationships_yet.json"))
+      end
+
+      it 'returns json with waiting status' do
+        get_request
+        expect(response.parsed_body['status']).to eq 'waiting'
+      end
+    end
+
+    context 'when relationship is expired' do
+      let(:reject_request_stub) { stub_request(:put, "#{connector_api_url}/Relationships/RELoi9IL4adMbj92K8dn/Changes/RCHNFJ9JD2LayPxn79nO/Reject") }
+
+      before do
+        stub_request(:get, "#{connector_api_url}/Relationships?status=Pending")
+          .to_return(body: File.read("#{mockfiles_path}/relationship_expired.json"))
```

Comment:
You could als try using the [`file_fixture` method](https://api.rubyonrails.org/v7.1.2/classes/ActiveSupport/Testing/FileFixtures.html) and placing the files somewhere under `spec/fixtures/files`.

Reply:
Yes, that looks better :+1: